src/libraries/System.Net.Quic/tests/FunctionalTests/QuicConnectionTests.cs

```diff
@@ -56,6 +56,149 @@ public async Task AcceptStream_ConnectionAborted_ByClient_Throws()
                     Assert.Equal(ExpectedErrorCode, ex.ErrorCode);
                 });
         }
+
+        private static async Task DoWrites(QuicStream writer, int writeCount)
+        {
+            for (int i = 0; i < writeCount; i++)
+            {
+                await writer.WriteAsync(new byte[1]);
+            }
+        }
+
+        private static async Task DoReads(QuicStream reader, int readCount)
+        {
+            for (int i = 0; i < readCount; i++)
+            {
+                int bytesRead = await reader.ReadAsync(new byte[1]);
+                Assert.Equal(1, bytesRead);
+            }
+        }
+
+        [Theory]
+        // [InlineData(0)]  // Fails with TimeoutException -- needs investigation
+        [InlineData(1)]
+        [InlineData(10)]
+        public async Task CloseAsync_WithOpenLocalStream_LocalStreamFailsWithQuicOperationAbortedException(int writesBeforeClose)
+        {
+            if (typeof(T) == typeof(MockProviderFactory))
+            {
+                return;
+            }
+
+            using var sync = new SemaphoreSlim(0);
+
+            await RunClientServer(
+                async clientConnection =>
+                {
+                    using QuicStream clientStream = clientConnection.OpenBidirectionalStream();
+                    await DoWrites(clientStream, writesBeforeClose);
+
+                    // Wait for peer to receive data 
+                    await sync.WaitAsync();
+
+                    await clientConnection.CloseAsync(ExpectedErrorCode);
+
+                    await Assert.ThrowsAsync<QuicOperationAbortedException>(async () => await clientStream.ReadAsync(new byte[1]));
+                    await Assert.ThrowsAsync<QuicOperationAbortedException>(async () => await clientStream.WriteAsync(new byte[1]));
+
+                    sync.Release();
+                },
+                async serverConnection =>
+                {
+                    using QuicStream serverStream = await serverConnection.AcceptStreamAsync();
+                    await DoReads(serverStream, writesBeforeClose);
+
+                    sync.Release();
+
+                    // Wait for peer to fail stream
+                    await sync.WaitAsync();
```

Comment:
Could you please also check server's side exceptions on the stream for all these tests?

Reply:
Actually, the second CloseAsync test below is effectively testing the reverse. So I can just merge these into a single test.